homeassistant/components/p1_monitor/__init__.py

```diff
@@ -0,0 +1,91 @@
+"""The P1 Monitor integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+
+from p1monitor import P1Monitor, Phases, Settings, SmartMeter
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_TIME_BETWEEN_UPDATE,
+    DEFAULT_TIME_BETWEEN_UPDATE,
+    DOMAIN,
+    LOGGER,
+    SERVICE_PHASES,
+    SERVICE_SETTINGS,
+    SERVICE_SMARTMETER,
+)
+
+PLATFORMS = (SENSOR_DOMAIN,)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up P1 Monitor from a config entry."""
+    hass.data.setdefault(DOMAIN, {}).setdefault(entry.entry_id, {})
+
+    client = P1Monitor(host=entry.data[CONF_HOST])
```

Comment:
You should be supplying the home assistant client http session to the monitor object (async_get_clientsession)

Reply:
I have no idea what you mean, can you make it a little clearer for me?

Reply:
Look at how you create the object in the config flow. You pass in a session object.

Reply:
Changed it, should be fine now?